proposed/links.md

```diff
@@ -87,23 +124,10 @@ interface LinkInterface
     /**
      * Returns a list of attributes that describe the target URI.
      *
-     * The list should be specified as a key-value list.
-     *
-     * There is no formal registry of the values that are allowed here, and
-     * validity of values is dependant on context.
-     *
-     * Common values are 'hreflang', 'title', and 'type'. Implementors
-     * embedding a serialized version of a link are responsible for only
-     * encoding the values they support.
-     *
-     * Any value that appears that is not valid in the context in which it is
-     * used should be ignored.
-     *
-     * Some attributes, (commonly hreflang) may appear more than once in their
-     * context. Attributes such as those may be specified as an array of
-     * strings.
-     *
-     * @return string[]
+     * @return array
```

Comment:
I considered that.  However, since attributes are always going to be fairly primitive, I don't think a traversable is necessary here.  The reason to use a traversable is when the dataset being returned could be very large or very slow to generate.  That could be the case for a set of Links, as they could be produced off of lazy DB queries, for instance, or the objects returned could be far more elaborate than just a trivial struct implementation of LinkInterface.  For attributes, since we're just returning, really, an array of strings (or an array of arrays of strings), neither of those situations applies.